### DISCUSSION TITLE
Issue With New Update

### DISCUSSION THREAD

**Discussion by cealagar** [giuliopons/adadmin#55] [General]
Hi! What could be the possible issue with this one after I updated to the latest version?

Fatal error: Uncaught TypeError: str_replace(): Argument #2 ($replace) must be of type string when argument #1 ($search) is a string in /home/<XXXXX>/src/componenti/7banner/_include/banner.class.php:1291 Stack trace: #0 /home/<XXXXX>src/componenti/7banner/_include/banner.class.php(1291): str_replace('##se_os##', Array, 'INSERT into 7ba...') #1 /home/<XXXXX>/src/componenti/7banner/index.php(175): Banner->updateAndInsert(Array, Array) #2 {main} thrown in /home/<XXXXX>/src/componenti/7banner/_include/banner.class.php on line 1291

This triggers when adding or updating banners.

PHP version: 8.0

**Comment by giuliopons** [comment]
I'm sorry I think that in this version 3.995 I've added also a bug in systems with php 8.  :-(

The problem is due to operating system checkboxes, I'll fix it this
evening. It didn't happen in my installation with php 7.4.

Giulio



Il dom 20 nov 2022, 10:23 Jeff Alagar ***@***.***> ha scritto:

**Comment by cealagar** [comment]
No worries kinda think it has something to do with the OS targeting feature too :)

Yup, I'm on PHP8. Thanks!

**Reply by giuliopons** [reply]
Just uploaded the fixed version on Codecanyon.
